pkg/stanza/operator/helper/emitter.go

```diff
@@ -147,9 +145,9 @@ func (e *LogEmitter) flusher(ctx context.Context) {
 			if oldBatch := e.makeNewBatch(); len(oldBatch) > 0 {
 				e.flush(context.Background(), oldBatch)
 			}
-		case <-ctx.Done():
+		case <-e.closeChan:
 			// flush currently batched entries
-			for oldBatch := e.makeNewBatch(); len(oldBatch) > 0; oldBatch = e.makeNewBatch() {
+			if oldBatch := e.makeNewBatch(); len(oldBatch) > 0 {
```

Comment:
As I mentioned in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34638#discussion_r1753839358, I believe it's fine to only do this once.